### PR TITLE
Support custom equality function

### LINKED ISSUE #19: Support snapshot style testing with html-looks-like
@staltz's new `html-looks-like` library looks like a lot of fun.

I want to be able to use it with @cycle/time.

https://github.com/staltz/html-looks-like

In order to do this, we might need to support custom assertion functions. This is probably an overall positive step for flexibility, and brings us more in line with how RxJS works.

Here's an example of what usage might look like:

```js
import {mockDOMSource, makeHTMLDriver} from '@cycle/dom';
import htmlLooksLike from 'html-looks-like';

import {Counter} from '../src/counter';

const htmlDriver = makeHTMLDriver();

function toHTML (vtree$) {
  return htmlDriver(vtree$).elements();
}

describe('Counter', () => {
  it('increments and decrements in response to clicks', (done) => {
    const Time = mockTimeSource();

    const expectedHTML = (count) => `
      <div class="counter">
        {{ ... }}
        <div class="count">Count: ${count}</div>
        {{ ... }}
      </div>
    `;

    const addClick$      = Time.diagram(`---x--x-------x--x--|`);
    const subtractClick$ = Time.diagram(`---------x----------|`);
    const expectedHTML$  = Time.diagram(`0--1--2--1----2--3--|`).map(expectedHTML);

    const DOM = mockDOMSource({
      '.add': {
        click: addClick$
      },

      '.subtract': {
        click: subtractClick$
      },
    });

    const counter = Counter({DOM});

    const HTML$ = counter.DOM.compose(toHTML);

    Time.assertEqual(HTML$, expectedHTML$, htmlLooksLike.bool)

    Time.run(done);
  });
});
```

### PR DESCRIPTION
**Pull request by raquelxmoss**
Closes #19 by supporting custom equality functions such as `html-looks-like`.

TODOs:
- [X] Remove unnecessary dependencies & any other clean up
- [X] Improve how the error messages display
- [x] Test that failures from custom equality functions are raised & displayed
- [x] Write documentation

### CODE REVIEW

**Comment by staltz** [comment]
`html-looks-like` ❤️

**Comment by jvanbruegge** [comment]
@Widdershin can you publish this as an alpha release? I would love to use this in my talk

File: README.md [hunk 1]
```diff
@@ -408,7 +408,7 @@ Time.run();
 // baz
 ```
 
-#### `assertEqual(actualStream, expectedStream, done)`
+#### `assertEqual(actualStream, expectedStream, comparator = assert.deepEqual)`
```
Comment (Widdershin) [inline README.md]: 👍 woops

File: README.md [hunk 2]
```diff
@@ -431,6 +433,41 @@ Time.assertEqual(
 Time.run(err => console.error(err));
 ```
 
+You can optionally pass a custom comparator function. This is useful if you want to do things like testing your DOM with tools such as [html-looks-like](https://github.com/staltz/html-looks-like).
+
```
Comment (Widdershin) [inline README.md]: Can we document that the comparator function can either return a bool or throw an error if something is wrong? This is useful as it allows the comparator to provide a custom error message.

**Comment by Widdershin** [comment]
Released `v0.8.0` with these changes.

**Comment by jvanbruegge** [comment]
Amazing

**Pull request merged**